io-engine/src/subsys/registration/registration_grpc.rs

```diff
@@ -66,10 +67,6 @@ pub struct Registration {
     config: Configuration,
     /// Registration client
     client: registration_client::RegistrationClient<tonic::transport::Channel>,
-    /// Receive channel for messages and termination
-    rcv_chan: async_channel::Receiver<()>,
-    /// Termination channel
-    fini_chan: async_channel::Sender<()>,
```

Comment:
Can we keep this and do the same for grpc.
Then everything can be triggered via the existing signal handler rather than spreading multiple signal handlers around which is looking a bit untidy.

Reply:
We cannot control the sequence in that case. The fini seems to get called at the last, which we don't want.

Reply:
We can simply call the inner of the ini directly from above

Reply:
Yeah, we would need to do the same for grpc server then as well. Will do it.